r/work.R

```diff
@@ -159,8 +158,27 @@ create_app <- function(last_modified, data, fpath) {
     return(app)
 }
 
-experiment_id <- Sys.getenv("EXPERIMENT_ID", unset = "e52b39624588791a7889e39c617f669e")
-message(paste("Welcome to Biomage R worker, experiment id", experiment_id))
+repeat {
+    label_path <- "/etc/podinfo/labels"
+    experiment_id <- NA
+
+    if(file.exists(label_path)) {
+        labels <- read.csv(label_path, sep="=", row.names=1, header=FALSE)
+        experiment_id <- labels["experimentId", ]
+    }
+
+    if(is.na(experiment_id)) {
+        experiment_id <- Sys.getenv("EXPERIMENT_ID", unset = NA)
+    }
+    
+    if(is.na(experiment_id)) {
+        message("No experiment ID label set yet, waiting...")
+        Sys.sleep(5)
```

Comment:
why is this in a separate if statements that are the same, rather than under 1 if statement?

Reply:
Essentially, we read from the file. If the file does not contain experiment_id, we read the env variable. If env variable also does not contain it, we're waiting.

Reply:
indeed, makes sense. Let's keep it this way